src/bp/ui-studio/src/web/views/OneFlow/sidePanel/form/SaySomethingForm.tsx

```diff
@@ -0,0 +1,244 @@
+    },
+    {
+      icon: 'trash',
+      label: 'Delete',
+      action: props?.onDeleteSelectedElements,
+      className: MoreOptionsStyles.delete
+    }
+  ]
+
+  const goThroughObjectAndLeaveOutIndex = (properties, indexToRemove) => {
+    const returnObject = {}
+
+    Object.keys(properties).forEach(key => {
+      if (key !== indexToRemove) {
+        returnObject[key] =
+          Object.prototype.toString.call(properties[key]) === '[object Object]'
+            ? goThroughObjectAndLeaveOutIndex(properties[key], indexToRemove)
+            : properties[key]
+      }
+    })
+
+    return returnObject
+  }
+
+  const removeDescriptions = json => {
+    const { properties, ...leftover } = json
+
+    const newProperties = goThroughObjectAndLeaveOutIndex(properties, 'description')
+
+    return { properties: newProperties, ...leftover }
+  }
+
+  const getCurrentCategory = () => {
+    if (!contentType || !categories) {
+      return
+    }
+
+    const {
+      schema: {
+        json: { description, title, ...json },
+        ...schema
+      },
+      ...category
+    } = categories?.find(cat => cat.id === contentType)
+
+    // just a way to remove the descriptions since we don't want it in the sidebar form, but still want it in the CMS
+    return { ...category, schema: { json: removeDescriptions(json), ...schema } }
+  }
+
+  const currentCategory = getCurrentCategory()
+
+  const handleEdit = event => {
+    if (!_.isEqual(event.formData, props.formData)) {
+      props.updateNode({
+        formData: event.formData
+      })
+    }
+  }
+
+  return (
+    <Fragment>
+      <div className={style.formHeader}>
+        <h4>Say Something</h4>
+        <MoreOptions show={showOptions} onToggle={setShowOptions} items={moreOptionsItems} />
+      </div>
+      <label className={style.fieldWrapper}>
+        <span className={style.formLabel}>Node Name</span>
+        <EditableInput
+          readOnly={readOnly}
+          value={currentFlowNode.name}
+          className={style.textInput}
+          onChanged={renameNode}
+          transform={transformText}
+        />
+      </label>
+      <div className={style.fieldWrapper}>
+        <span className={style.formLabel}>Content Type</span>
+        {categories && (
+          <Dropdown
+            className={style.formSelect}
+            items={categories.map(cat => ({ value: cat.id, label: cat.title }))}
+            defaultItem={contentType}
+            rightIcon="caret-down"
+            onChange={option => {
+              handleContentTypeChange(option.value)
+            }}
+          />
+        )}
+      </div>
+
+      {currentCategory && (
+        <ContentForm
+          schema={currentCategory?.schema.json}
+          uiSchema={currentCategory?.schema.ui}
+          formData={currentFlowNode.formData}
+          isEditing={true}
+          onChange={handleEdit}
+        />
+      )}
+    </Fragment>
+  )
+}
+
+const mapStateToProps = state => ({
```

Comment:
(state: RootReducer) so you have typings & autocompletion when using the ReturnType to type them